pkg/kubectl/apply/strategy/replace_visitor.go

```diff
@@ -53,7 +55,14 @@ func (v replaceStrategy) MergeType(e apply.TypeElement) (apply.Result, error) {
 
 // MergePrimitive returns a result by merging the recorded, local and remote values
 // - replacing the remote value with the local value
+// If FailOnConflict option is enabled, throw error when conflicts occure in both recorded
+// and remote values.
 func (v replaceStrategy) MergePrimitive(e apply.PrimitiveElement) (apply.Result, error) {
+	if v.options.FailOnConflict && e.HasRecorded() && e.HasRemote() &&
```

Comment:
This is a really complex conditional.  How can we structure this to make it clear what the intent is?